libs/api/ebpf_api.cpp

```diff
@@ -1475,6 +1481,169 @@ _get_next_map_to_create(std::vector<ebpf_map_t*>& maps)
     return nullptr;
 }
 
+ebpf_result_t
+ebpf_object_open(
+    _In_z_ const char* path,
+    _In_opt_z_ const char* object_name,
+    _In_opt_ const ebpf_program_type_t* program_type,
+    _In_opt_ const ebpf_attach_type_t* attach_type,
+    _Outptr_ struct bpf_object** object,
+    _Outptr_result_maybenull_z_ const char** error_message) noexcept
+{
+    *error_message = nullptr;
+
+    ebpf_object_t* new_object = new (std::nothrow) ebpf_object_t();
+    if (new_object == nullptr) {
+        return EBPF_NO_MEMORY;
+    }
+
+    ebpf_result_t result =
+        _initialize_ebpf_object_from_elf(path, object_name, program_type, attach_type, *new_object, error_message);
+    if (result != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    *object = new_object;
+    _ebpf_objects.emplace_back(*object);
+
+Done:
+    if (result != EBPF_SUCCESS) {
+        _clean_up_ebpf_object(new_object);
+    }
+    return result;
+}
+
+static ebpf_result_t
+_ebpf_object_create_maps(_Inout_ ebpf_object_t* object)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+
+    clear_map_descriptors();
+
+    // TODO: update ebpf_map_definition_t structure so that it contains flag and pinning information.
+    for (int count = 0; count < object->maps.size(); count++) {
+        ebpf_map_t* map = _get_next_map_to_create(object->maps);
+        if (map == nullptr) {
+            // Any remaining maps cannot be created.
+            result = EBPF_INVALID_OBJECT;
```

Comment:
When all the maps have been created, `_get_next_map_to_create` will return `nullptr`. So I think the result here should not be `EBPF_INVALID_OBJECT`?

Reply:
The `count < object->maps.size()` condition in line 1524 means this should not happen normally.  This path can only happen if there are maps left to create but they are not creatable (e.g., two outer maps that reference each other as the inner map), in which case this error is appropriate.